README.md

```diff
@@ -1,34 +1,149 @@
+%}
+```
+
+#### Delete 
+
+Delete an existing embedding with the `modules/openai/embeddings/delete` command:
+
+```
+{% function pos_embedding = 'modules/openai/commands/embeddings/delete', id: embedding.id %}
+```
+
+#### Read
+
+To leverage embeddings, you will need to perform a search operation through `modules/openai/queries/embeddings/search.liquid`. Typically, you'll want to sort the results by their relevance to the input embedding (for example, when a user enters a search query, which you transform into an embedding using the `fetch_embedding` command, and you want to retrieve the five most relevant embeddings):
+
+```
+  function related_embeddings = 'modules/openai/queries/embeddings/search', related_to: embedding, limit: 5
+```
+
+## Example application
+
+One typical use case for embeddings is to implement AI-enhanced search, which is also the first step in Retrieval-Augmented Generation (RAG). You can find an example implementation of this functionality in the `app` directory. It’s ready to be copied and customized for your application needs or further integrated with the LLM of your choice. This example demonstrates key concepts in action.
+
+### Script to Transform Existing Data into Embeddings
+
+You can find the script in `app/lib/commands/openai/commands_to_embeddings.liquid`. It fetches the existing [Pages](https://documentation.platformos.com/developer-guide/pages/pages), filters them according to your needs, and creates embeddings based on their content. This is intended for demonstration purposes; for instance, instead of creating embeddings for entire pages, you might choose to create embeddings for specific paragraphs, which could yield better results.
+
+The script can be run multiple times and will remove embeddings of pages that have been deleted since the last run, update embeddings for pages that have changed, and add new embeddings for newly added pages.
+
+The script assumes that each indexed page has configured [Metadata](https://documentation.platformos.com/developer-guide/pages/metadata) — `title` and `description`. These are required to efficiently render the [Search Page](#search-page) without needing to make another database query.
+
+### Search page
+
+The example search page is located at `app/views/pages/openai_search.liquid`. It serves primarily for demonstration purposes and intentionally violates some [best practices](https://documentation.platformos.com/developer-guide/modules/platformos-modules#how-to-use-devkit), such as separating business rules from the presentation layer. This design choice aims to present an all-in-one example of the entire workflow.
+
+You'll find there additional checks to ensure the module [setup](#setup) has been configured correctly. It supports the hCaptcha challenge to prevent spam requests and demonstrates how to use the [commands and queries](#functionality-provided-by-the-user-module) provided by the OpenAI module.
```

Comment:
```suggestion
The example search page includes additional checks to ensure the module [setup](#setup) has been configured correctly. It supports the hCaptcha challenge to prevent spam requests and demonstrates how to use the [commands and queries](#functionality-provided-by-the-user-module) provided by the OpenAI module.
```